### PR TITLE
Clean up doc build

### PR DESCRIPTION
**Pull request by whit537**
Good docs build cleanly!

### CODE REVIEW

**Comment by whit537** [comment]
```
(cloud-custodian) $ sphinx-build -W docs/source docs/build -q 2>&1 | less
(cloud-custodian) $
```

**Comment by whit537** [comment]
Ready for review. Note that this brings the API reference into the docs. Are we ready for that? If not we can turn off `sphinx-apidoc` (building w/o linking in the ToC throws a warning).

screen shot 2016-12-21 at 8 49 24 am

File: c7n/utils.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def type_schema(
                  inherits issues with additionalProperties and type enums.
      - aliases: additional names this type maybe called
      - required: list of required properties, by default 'type' is required
-     - **props: additional key value properties
+     - props: additional key value properties
```
Comment (scotwk) [inline c7n/utils.py]: Shouldn't the ** be escaped rather than deleting it?

**Comment by kapilt** [comment]
i'm fine with it for now, i just don't think auto module docs reflects end target state, where reference docs are not nesc a code docs but a policy authoring reference thats autogen'd from registered resource types and their respective filters/actions. but its a good step forward.

**Pull request merged**